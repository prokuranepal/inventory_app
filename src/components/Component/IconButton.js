import React from "react";
import {
    TouchableOpacity,
    TouchableNativeFeedback,
    Text,
    View,
    StyleSheet,
    Platform
} from "react-native";
import { Ionicons } from '@expo/vector-icons';

import Colors from '../../constants/Colors'
const iconButton = props => {

    let TouchableCmp = TouchableOpacity;
    if (Platform.OS === "android" && Platform.Version >= 21) {
        TouchableCmp = TouchableNativeFeedback;
    }
    return (
        <View style={{ width: "100%", alignItems: "center" }} >
            <TouchableCmp onPress={props.onPressHandler} data-test="touchableComp">
                <View style={styles.gridItem} >
                    <View style={{ ...styles.container, ...{ backgroundColor: props.color } }}>
                        <View style={styles.iconContainer}>
                            <Ionicons name={props.iconValue} data-test="iconComp" size={100} color={props.iconColor} />
                        </View>

                    </View>

                </View>
            </TouchableCmp>
        </View >
    );

    ;
};

const styles = StyleSheet.create({
    gridItem: {
        flex: 1,
        margin: 15,
        height: 120,
        width: "30%",
        borderRadius: 10,
        alignItems: "center",
        overflow: Platform.OS == "android" && Platform.Version >= 21 ? 'hidden' : 'visible',
        elevation: 1,
        alignItems: "center"

    },
    container: {
        flex: 1,
        borderRadius: 10,
        shadowColor: 'black',
        shadowOpacity: 0.26,
        shadowOffset: { width: 0, height: 2 },
        shadowRadius: 1,
        padding: 10,

    },

    iconContainer: {
        flex: 1,
        width: "100%",
        height: "100%",
        justifyContent: 'center',
        alignItems: 'center',
        opacity: 0.7


    },
});

export default iconButton;