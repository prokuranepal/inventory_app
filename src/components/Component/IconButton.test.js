import React from 'react';
import IconButton from './IconButton';
import {
    configure,
    shallow
} from 'enzyme';
import  {findByTestAttr, mockPlatform} from '../../_test_/componenets/test_function';
import EnzymeAdapter from 'enzyme-adapter-react-16';

configure({
    adapter: new EnzymeAdapter
})
const jest_function=jest.fn()



const prop_data = {
  iconValue:"ios-add",
  iconColor:"white",
  color:"green",
  onPressHandler: jest_function

}
const findById=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}

describe('Test for IconButton', () => {
  it("Snapshot test", () =>
  {
    const wrapper = shallow(<IconButton  {...prop_data} />)
    expect(wrapper).toMatchSnapshot()
  })
  it("Props in IconButton", () =>
       
        {
            const wrapper = shallow(<IconButton  {...prop_data} />)
            let iconComp= findByTestAttr(wrapper,"iconComp")
            expect(iconComp.prop('name')).toEqual('ios-add')
            expect(iconComp.prop('color')).toEqual('white')
        });
        it("Events in IconButton", () =>
       
        {
            const wrapper = shallow(<IconButton  {...prop_data} />)
            let touchableComp= findByTestAttr(wrapper,"touchableComp")
            touchableComp.props().onPress();
            expect(jest_function).toHaveBeenCalled()
        });

 });
