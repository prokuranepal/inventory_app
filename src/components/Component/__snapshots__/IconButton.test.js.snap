// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for IconButton Snapshot test 1`] = `ShallowWrapper {}`;
